resource/systemd/unit/unit.go

```diff
@@ -124,42 +86,42 @@ func (t *Unit) Check(r resource.Renderer) (resource.TaskStatus, error) {
 
 	if err != nil {
 		// Unit doesn't exist should be linked
-		if err.Error() == "LoadError: \"[\\\"org.freedesktop.DBus.Error.FileNotFound\\\", \\\"No such file or directory\\\"]\"" {
+		if err.Error() == loadError {
 			if shouldBeLinked {
+				fmt.Println("Here--------------\n\n")
 				// Check if file to be linked exist on disk
 				_, statErr := os.Stat(t.Name)
 				if statErr != nil {
-					return &resource.Status{
-						Status:       err.Error(),
-						WarningLevel: resource.StatusFatal,
-						Output:       []string{err.Error()},
-					}, err
+					t.TaskStatus = &resource.Status{
+						Level:  resource.StatusFatal,
+						Output: []string{statErr.Error()},
+					}
+					return t.TaskStatus, statErr
 				}
 				statusMsg := fmt.Sprintf("unit %q does not exist, will be linked", unitName)
 				diffMsg := fmt.Sprintf("unit %q does not exist", unitName)
 				diff := resource.TextDiff{Default: diffMsg, Values: [2]string{diffMsg, fmt.Sprintf("unit %q is linked", unitName)}}
-				return &resource.Status{
-					Status:       statusMsg,
-					WarningLevel: resource.StatusWillChange,
-					WillChange:   true,
-					Differences:  map[string]resource.Diff{unitName: diff},
-					Output:       []string{statusMsg, err.Error()},
-				}, nil
+				t.TaskStatus = &resource.Status{
+					Level:       resource.StatusWillChange,
+					Differences: map[string]resource.Diff{unitName: diff},
+					Output:      []string{statusMsg},
+				}
+				return t.TaskStatus, nil
 			} else if t.UnitFileState.Equal(systemd.UFSDisabled) {
 				// Consider it the same as it being disabled.
 				statusMsg := fmt.Sprintf("unit %q does not exist, considered disabled", unitName)
-				return &resource.Status{
-					Status:       statusMsg,
-					WarningLevel: resource.StatusWontChange,
-					Output:       []string{statusMsg},
-				}, nil
+				t.TaskStatus = &resource.Status{
+					Level:  resource.StatusWontChange,
+					Output: []string{statusMsg},
+				}
+				return t.TaskStatus, nil
 			}
 		}
-		status := &resource.Status{
-			WarningLevel: resource.StatusFatal,
-			Status:       err.Error(),
+		t.TaskStatus = &resource.Status{
+			Level:  resource.StatusFatal,
+			Output: []string{err.Error()},
 		}
-		return status, err
+		return t.TaskStatus, err
```

Comment:
the return value here should be `t`. Since you're embedding the TaskStatus, we can do lookups on the task fields as well.